doc/src/main/scala/org/bitcoins/doc/wallet/create-wallet.sc

```diff
@@ -0,0 +1,230 @@
+import org.bitcoins.rpc.config.BitcoindInstance
+import org.bitcoins.rpc.util.RpcUtil
+import org.bitcoins.testkit.rpc.BitcoindRpcTestUtil
+import org.bitcoins.wallet.db.WalletDbManagement
+import org.bitcoins.wallet.db.WalletDbConfig
+import org.bitcoins.wallet.config.WalletAppConfig
+
+import org.bitcoins.zmq.ZMQSubscriber
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters._
+import scala.concurrent._
+import scala.concurrent.duration.DurationInt
+import scala.util._
+/**
+* This is for example purposes only!
+  * This shows how to peer a bitcoin-s wallet
+  * with a bitcoind instance that is relaying
+  * information about what is happening on the blockchain
+  * to the bitcoin-s wallet.
+  *
+  * This is useful if you want more flexible signing
+  * procedures in the JVM ecosystem and more
+  * granular control over your utxos with
+  * popular databases like postgres, sqlite etc
+  */
+
+
+val logger = LoggerFactory.getLogger("org.bitcoins.doc.wallet.CreateWallet")
+val time = System.currentTimeMillis()
+//boiler plate config
+implicit val system = ActorSystem(s"wallet-scala-sheet-${time}")
+import system.dispatcher
+
+val chainDbConfig = ChainDbConfig.RegTestDbConfig
+val chainAppConfig = ChainAppConfig(chainDbConfig)
+implicit val chainParams = chainAppConfig.chain
+
+val walletDbConfig = WalletDbConfig.RegTestDbConfig
+val walletAppConfig = WalletAppConfig(walletDbConfig)
+
+val datadir = new File(s"/tmp/bitcoin-${time}/")
+val bitcoinConf = new File(datadir.getAbsolutePath + "/bitcoin.conf")
+
+logger.info(s"bitcoin.conf location=${bitcoinConf.getAbsolutePath}")
+datadir.mkdirs()
+bitcoinConf.createNewFile()
+
+val config = BitcoindRpcTestUtil.standardConfig
+val _ = BitcoindRpcTestUtil.writeConfigToFile(config,datadir)
+
+//construct bitcoind
+val instance = BitcoindInstance.fromConfig(config = config, datadir)
+val bitcoind = new BitcoindRpcClient(instance = instance)
+
+//start bitcoind, this may take a little while
+//generate 101 blocks so we have money in our wallet
+val bitcoindF = bitcoind.start().map(_ => bitcoind)
+
+//create a native chain handler for bitcoin-s
+val blockHeaderDAO: BlockHeaderDAO = BlockHeaderDAO(appConfig = chainAppConfig)
+val genesisHeader = BlockHeaderDbHelper.fromBlockHeader(
+  height = 0,
+  bh = chainAppConfig.chain.genesisBlock.blockHeader)
+
+
+val blockHeaderTableF = {
+  //drop regtest table if it exists
+  val dropTableF = ChainDbManagement.dropHeaderTable(chainDbConfig)
+
+  //recreate the table
+  val createdTableF = dropTableF.flatMap(_ => ChainDbManagement.createHeaderTable(chainDbConfig))
+
+  createdTableF
+}
+val createdGenHeaderF = blockHeaderTableF.flatMap(_ => blockHeaderDAO.create(genesisHeader))
+
+val chainF = createdGenHeaderF.map(h => Vector(h))
+
+val blockchainF = chainF.map(chain => Blockchain(chain))
+
+val chainHandlerF = blockchainF.map(blockchain => ChainHandler(blockHeaderDAO, chainAppConfig))
+
+val chainApi101BlocksF = sync(chainHandlerF, 101)
```

Comment:
I'm getting an error "Wrong forward reference" for `sync` here. It's in Intellij only so probably not a big deal, but might make sense to move it to before its used?